src/hooks/use-site-sync-management.ts

```diff
@@ -0,0 +1,92 @@
+import { useState, useEffect, useCallback } from 'react';
+import { getIpcApi } from '../lib/get-ipc-api';
+import { useAuth } from './use-auth';
+import { SyncSite, useFetchWpComSites } from './use-fetch-wpcom-sites';
+import { useSiteDetails } from './use-site-details';
+
+const useSiteSyncManagement = () => {
+	const [ connectedSites, setConnectedSites ] = useState< SyncSite[] >( [] );
+	const { isAuthenticated } = useAuth();
+	const { syncSites } = useFetchWpComSites( connectedSites );
+	const { selectedSite } = useSiteDetails();
+	const localSiteId = selectedSite?.id;
+
+	const loadConnectedSites = useCallback( async () => {
+		if ( ! localSiteId ) {
+			setConnectedSites( [] );
+			return;
+		}
+
+		try {
+			const sites = await getIpcApi().getConnectedWpcomSites( localSiteId );
+			setConnectedSites( sites );
+		} catch ( error ) {
+			console.error( 'Failed to load connected sites:', error );
+			setConnectedSites( [] );
+		}
+	}, [ localSiteId ] );
+
+	useEffect( () => {
+		if ( isAuthenticated ) {
+			loadConnectedSites();
+		}
+	}, [ isAuthenticated, loadConnectedSites ] );
+
+	const connectSite = useCallback(
+		async ( site: SyncSite ) => {
+			if ( ! localSiteId ) {
+				return;
+			}
+			try {
+				const stagingSites = site.stagingSiteIds
+					.map( ( id ) => syncSites?.find( ( s ) => s.id === id ) )
```

Comment:
Do we need this question mark?

```suggestion
					.map( ( id ) => syncSites.find( ( s ) => s.id === id ) )
```

Reply:
Adjusted in https://github.com/Automattic/studio/pull/610/commits/bef0f2fde09a8ae6740fcdbcb5d29705403f83f3